modules/dhp_connected_devices/app/controllers/dhp_connected_devices/veteran_device_records_controller.rb

```diff
@@ -3,8 +3,17 @@
 module DhpConnectedDevices
   class VeteranDeviceRecordsController < ApplicationController
     def index
-      device_records = Device.veteran_device_records(@current_user)
-      render json: VeteranDeviceRecordSerializer.serialize(device_records[:active], device_records[:inactive])
+      if @current_user.icn.present?
```

Comment:
We can use the nil safe `&.` operator here just in case `@current_user` is nil
```suggestion
      if @current_user&.icn.present?
```

even slightly cleaner we can do 
```suggestion
      render json: { connectionAvailable: false } if @current_user&.icn.present?
      ... # rest of the code goes here
```

Reply:
Going with the first suggestion. Rail's render doesn't terminate execution of the action and won't accept two renders on the same block level.

Reply:
Rearranged the conditionals to check for quick exit condition (if @current_user&.icn.blank?) first